File: ChessClubBurgas64.Data/Models/Announcement.cs
#nullable disable
using System.ComponentModel.DataAnnotations;

namespace ChessClubBurgas64.Data.Models
{
    public class Announcement
    {
        public Announcement()
        {
            Images = new HashSet<Image>();
        }

        [Key]
        public Guid Id { get; set; }

        public string Title { get; set; }

        public DateTime DateCreated { get; set; } = DateTime.UtcNow;

        public DateTime DateUpdated { get; set; } = DateTime.UtcNow;

        public string Description { get; set; }

        public string Text { get; set; }

        public virtual ICollection<Image> Images { get; set; }
    }
}

File: ChessClubBurgas64.Web/ClientApp/src/app/models/announcement.ts
export interface Announcement {
    id: string;
    title: string;
    dateCreated: Date | null;
    dateUpdated: Date | null;
    description: string;
    text: string;
    mainImageUrl: string;
    images: Image[]
}

export class AnnouncementFormValues
{
  id?: string = undefined;
  title: string = '';
  description: string = '';
  text: string = '';
  mainImageUrl: string = '';
  
  constructor(announcement?: AnnouncementFormValues) {
    if (announcement) {
      this.id = announcement.id;
      this.title = announcement.title;
      this.description = announcement.description;
      this.text = announcement.text;
      this.mainImageUrl = announcement.mainImageUrl;
    }
  }
}

export class Announcement implements Announcement {
  constructor(init?: AnnouncementFormValues) {
    Object.assign(this, init);
  }
}

export interface Image {
  id: string;
  url: string;
  isMain: boolean;
}
